### PR TITLE
Applying this commit will fix the "tx.wait is not a function error" i…

### PR DESCRIPTION
**Pull request by tracyarciaga**
## Description of Changes

- Add try catch block in withdraw func in HaloHaloWithdrawPanel.tsx
- Add try catch block in deposit func in HaloDepositPanel.tsx

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-243)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/pages/HaloHalo/HaloDepositPanel.tsx [hunk 1]
```diff
@@ -165,18 +165,21 @@ export default function CurrencyInputPanel({
...
-      amount = maxDepositAmountInput
-    } else {
-      amount = formatToBalance(depositValue, decimals)
+      let amount: BalanceProps | undefined
+      if (maxSelected) {
+        amount = maxDepositAmountInput
+      } else {
+        amount = formatToBalance(depositValue, decimals)
+      }
+      const tx = await enter(amount)
```
Comment (schystz) [inline src/pages/HaloHalo/HaloDepositPanel.tsx:178]: @tracyarciaga I think we just need lines 178 & 179 inside the try block, other codes above can remain outside the try-catch statement.

File: src/pages/HaloHalo/HaloHaloWithdrawPanel.tsx [hunk 2]
```diff
@@ -162,18 +162,21 @@ export default function HaloHaloWithdrawPanel({
...
-      amount = maxWithdrawAmountInput
-    } else {
-      amount = formatToBalance(withdrawValue, decimals)
+      let amount: BalanceProps | undefined
+      if (maxSelected) {
+        amount = maxWithdrawAmountInput
+      } else {
+        amount = formatToBalance(withdrawValue, decimals)
+      }
+      const tx = await leave(amount)
```
Comment (schystz) [inline src/pages/HaloHalo/HaloHaloWithdrawPanel.tsx:175]: @tracyarciaga I think we just need lines 175 & 176 inside the try block, other codes above can remain outside the try-catch statement.

**Review by schystz — COMMENTED** [review]
Just a couple of comments @tracyarciaga. And I think this branch needs to be synced to develop branch as well to remove the lint warnings.

File: src/pages/HaloHalo/HaloHaloWithdrawPanel.tsx (hunk 2 above)
Comment (tracyarciaga) [reply src/pages/HaloHalo/HaloHaloWithdrawPanel.tsx:175]: This is done

File: src/pages/HaloHalo/HaloDepositPanel.tsx (hunk 1 above)
Comment (tracyarciaga) [reply src/pages/HaloHalo/HaloDepositPanel.tsx:178]: done

**Review by ernestpascual — APPROVED** [review]

**Review by schystz — APPROVED** [review]
Lgtm!

**Pull request merged**